File: GUISM_SERVER/source/caminho/Caminho.java
package caminho;

import especializacao.Especializacao;
import habilidade.Habilidade;
import habito.Habito;
import kikaha.jdbi.serializers.Column;
import kikaha.jdbi.serializers.Entity;
import lombok.Data;

import java.util.Set;

@Data
@Entity
public class Caminho {

    @Column
    Long idCaminho;

    @Column
    String nomeCaminho;

    @Column
    String descCaminho;

   Set<Habilidade> habilidades;

    Set<Especializacao> especializacoes;

    Set<Habito> habitos;
}

File: src/components/caminho/Caminho.ts
import Habilidade from "../habilidade/Habilidade";
import Especializacao from "../especializacao/Especializacao";
import Habito from "../habito/Habito";

export default  interface Caminho {
    idCaminho: number;
    nomeCaminho: string;
    descCaminho: string;
    nivelCaminho: number;
    habilidades: Habilidade[];
    especializacoes: Especializacao[];
    habitos: Habito[];
}
